common/metrics/otel_metrics_handler.go

```diff
@@ -37,13 +38,36 @@ import (
 )
 
 // otelMetricsHandler is an adapter around an OpenTelemetry [metric.Meter] that implements the [Handler] interface.
-type otelMetricsHandler struct {
-	l           log.Logger
-	tags        []Tag
-	provider    OpenTelemetryProvider
-	excludeTags excludeTags
-	catalog     catalog
-}
+type (
+	otelMetricsHandler struct {
+		l            log.Logger
+		set          attribute.Set
+		provider     OpenTelemetryProvider
+		tagConverter tagConverter
+		catalog      catalog
+		gauges       *sync.Map // string -> *gaugeAdapter. note: shared between multiple otelMetricsHandlers
```

Comment:
We've opened up a new class of OOM errors here if our gauge cardinality explodes. Can we have a safety valve for the size of this map?

Reply:
Does tally have any similar guards? Or does the rest of the otel library itself? I don't think there's justification for adding it here and not anywhere else in the metrics stack